ya-erigon-runtime/src/main.rs

```diff
@@ -1,36 +1,53 @@
 use futures::channel::oneshot;
 use futures::FutureExt;
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
 use serde::{Deserialize, Serialize};
+use std::iter;
 use std::path::PathBuf;
 use std::process::Stdio;
 use std::sync::atomic::AtomicU64;
 use std::sync::atomic::Ordering::Relaxed;
 use tokio::process::{Child, Command};
 use ya_runtime_sdk::*;
 
+const AUTH_ERIGON_USER: &str = "erigolem";
 const ERIGON_BIN: &str = "tg";
 const RPCDAEMON_BIN: &str = "rpcdaemon";
 
 //TODO: Make parameter list configurable for further extendability (Erigon & Rpcdaemon)
-const RPCDAEMON_PARAMS: &[&str; 12] = &[
+const RPCDAEMON_PARAMS: &[&str; 10] = &[
     "--private.api.addr",
     "localhost:9090",
     "--http.addr",
-    "0.0.0.0",
+    "127.0.0.1",
     "--http.port",
     "8545",
     "--http.vhosts",
     "*",
-    "--http.corsdomain",
-    "*",
     "--http.api",
     "eth,debug,net,trace,web3,tg",
 ];
 
-#[derive(Default, Deserialize, Serialize)]
+#[derive(Deserialize, Serialize)]
 pub struct ErigonConf {
-    public_addr: Option<String>,
-    data_dir: Option<String>,
+    public_addr: String,
+    data_dir: String,
+    passwd_tool_path: String,
+    passwd_file_path: String,
+    password_default_length: usize,
+}
+
+impl Default for ErigonConf {
+    fn default() -> Self {
+        ErigonConf {
+            public_addr: String::from("http://erigon.localhost:8545"),
+            data_dir: String::from("/data/turbo-geth"),
+            passwd_tool_path: String::from("/usr/bin/htpasswd"),
```

Comment:
Not sure if the default should be a full path or rather taken from `PATH`. Would `Command::new(...).spawn()` work if we used simply `htpasswd`?

Reply:
Yes I've changed it also in `Defaults` impl